### PR TITLE
ENH add mamba solver timeout at very high level

### PR DESCRIPTION
**Pull request by beckermr**
I think this will work.

### CODE REVIEW

**Comment by beckermr** [comment]
ok @CJ-Wright @wolfv I think this will work for timing out the solver for now. Can you all check the process handling and do some CR?

**Comment by beckermr** [comment]
also @chrisburr for viz and CR

File: conda_forge_tick/mamba_solver.py [hunk 1]
```diff
@@ -539,9 +488,19 @@ def clean():
     return repodata.channel_url
 
 
+def _func(feedstock_dir, additional_channels, conn):
+    res = _is_recipe_solvable(
+        feedstock_dir,
+        additional_channels=additional_channels,
+    )
+    conn.send(res)
+    conn.close()
```
Comment (chrisburr) [inline conda_forge_tick/mamba_solver.py:497]: This needs to catch exceptions and pass them back to the parent:
```suggestion
    try:
        res = _is_recipe_solvable(
            feedstock_dir,
            additional_channels=additional_channels,
        )
        conn.send(res)
    except Exception as e:
        conn.send(e)
    finally:
        conn.close()
```

File: conda_forge_tick/mamba_solver.py [hunk 2]
```diff
@@ -565,6 +530,47 @@ def is_recipe_solvable(
...
+        from multiprocessing import Process, Pipe
+
+        parent_conn, child_conn = Pipe()
+        p = Process(
+            target=_func,
+            args=(feedstock_dir, additional_channels, child_conn),
+        )
+        p.start()
+        if parent_conn.poll(timeout):
+            res = parent_conn.recv()
```
Comment (chrisburr) [inline conda_forge_tick/mamba_solver.py:543]:
```suggestion
            res = parent_conn.recv()
            if isinstance(res, Exception):
                TODO....
```

Comment (beckermr) [reply conda_forge_tick/mamba_solver.py:547]: done.

**Comment by beckermr** [comment]
Thanks @chrisburr! Ready for another look!

**Review by chrisburr — APPROVED** [review]

**Pull request merged**